converters/schema.ts

```diff
@@ -0,0 +1,75 @@
+import { ClarityDataSchema, ObjectType } from "../declarations/clarity";
+import { hashCode } from "./../src/utils";
+
+export class SchemaManager {
+
+  private schemas = {};
+
+  public getSchema(schemaHash: string): ClarityDataSchema {
+    return this.schemas[schemaHash];
+  }
+
+  public addSchema(schema: ClarityDataSchema): boolean {
+    let schemaHash = hashCode(JSON.stringify(schema));
```

Comment:
>hashCode [](start = 21, length = 8)

Why do we need hashcode for schema, we can probably prevent this indirection by storing schema string itself as key. E.g. "index,layout.x,layout.y,..." is unique in itself.